### PR TITLE
Add some HOS service

### PR DESCRIPTION
**Pull request by HorrorTroll**
Add:
- [x] nifm IFriendService: 10600 (DeclareOpenOnlinePlaySession)
- [x] irs IIrSensorServer: 319 (ActivateIrsensorWithFunctionLevel)
- [x] nvdrv INvDrvServices: 9 (DumpGraphicsMemoryInfo)

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Irs/IIrSensorServer.cs [hunk 1]
```diff
@@ -40,5 +41,16 @@ public long DeactivateIrsensor(ServiceCtx context)
 
             return 0;
         }
+
+        // ActivateIrsensorWithFunctionLevel(nn::applet::AppletResourceUserId, nn::irsensor::PackedFunctionLevel, pid)
+        public long ActivateIrsensorWithFunctionLevel(ServiceCtx context)
+        {
+            long appletResourceUserId = context.RequestData.ReadInt64();
+            int  packedFunctionLevel  = context.RequestData.ReadInt32();
```
Comment (HorrorTroll) [inline Ryujinx.HLE/HOS/Services/Irs/IIrSensorServer.cs]: Unknown this is right or not, need better reverse engineering work.

**Comment by jduncanator** [comment]
We've made a conscious decision as a team to avoid merging new stubs for services as much as possible going forward, unless the function is completely unimplementable or relatively benign. I'm happy to accept the stubs for the IR services and DumpGraphicsMemoryInfo, as well as the partial stub for DeclareOpenOnlinePlaySession, but NVHOST_IOCTL_CTRL_EVENT_KILL needs to be properly reversed and implemented, and CheckFreeCommunicationPermission needs to have it's return codes properly implemented before I can accept this in a PR. 

Feel free to remove those two implementations, and I'll review the rest, but I can't merge unless those two changes are implemented correctly.

Perhaps it's best to break each change out into it's own PR. It makes it easier to revert or investigate issues when you don't have multiple unrelated changes merged in a single PR.

**Comment by gdkchan** [comment]
fwiw EVENT_REGISTER is also stubbed, and its not possible  to implement EVENT_KILL properly without implementing EVENT_REGISTER first.

**Comment by HorrorTroll** [comment]
Okay! Thanks for review, i will take a look a code later.

**Comment by HorrorTroll** [comment]
All done

**Comment by HorrorTroll** [comment]
All done, ready for a review

File: Ryujinx.HLE/HOS/Services/Pctl/IParentalControlService.cs [hunk 2]
```diff
@@ -38,4 +38,4 @@ public long Initialize(ServiceCtx context)
             return 0;
         }
     }
-}
\ No newline at end of file
+}
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/Pctl/IParentalControlService.cs]: I don't think it should be part of the PR :/

Comment (HorrorTroll) [reply Ryujinx.HLE/HOS/Services/Pctl/IParentalControlService.cs]: i was deleted pctl 1001, because it's was merged to master

**Comment by HorrorTroll** [comment]
Rework at here: https://github.com/Ryujinx/Ryujinx/pull/679!

Sorry :/

**Pull request closed without merging**